### PR TITLE
Description editing feature

### PR DESCRIPTION
**Pull request by jozsefsallai**
Description editing functionality + UI + tests

### CODE REVIEW

File: api/tests/factory.js [hunk 1]
```diff
@@ -37,7 +37,8 @@ factory.define('audio', Audio, {
   originalName: () => `${factory.chance('word')()}.mp3`,
   size: 10000,
   mimetype: 'audio/mpeg',
-  url: factory.seq('Audio.url', n => `${factory.chance('word')()}${n}`)
+  url: factory.seq('Audio.url', n => `${factory.chance('word')()}${n}`),
+  description: null
```
Comment (austindebruyn) [inline api/tests/factory.js]: `factory.chance('sentence')`?

File: ui/src/components/dashboard/audio_list_item.vue [hunk 2]
```diff
@@ -7,30 +7,40 @@
...
-        span.fa.fa-eye-slash(v-if='!audio.visible')
-      .description {{ display_description }}
+          input.edit-description-input(
+            v-model='edit_description_input_value'
+            type='text'
+            placeholder='Description'
+            :disabled='loading'
+            @click='handle_edit_description_input_click'
+          )
+      .audio-info(v-else=true)
```
Comment (austindebruyn) [inline ui/src/components/dashboard/audio_list_item.vue]: looks like `.audio-info` is unused?

File: api/tests/factory.js (hunk 1 above)
Comment (jozsefsallai) [reply api/tests/factory.js]: whoops, my bad! will fix that soon

File: ui/src/components/dashboard/audio_list_item.vue (hunk 2 above)
Comment (austindebruyn) [reply ui/src/components/dashboard/audio_list_item.vue]: what about `div(v-else=true)`? class name is unneeded if nothing references it and it has no styles

**Comment by austindebruyn** [comment]
@jozsefsallai oh was there meant to be a migration in here somewhere

File: ui/src/components/dashboard/audio_list_item.vue (hunk 2 above)
Comment (jozsefsallai) [reply ui/src/components/dashboard/audio_list_item.vue]: yeah, guess i don't like plain, naked divs haha
will modify that in a new commit

**Comment by jozsefsallai** [comment]
@austindebruyn yeah, that was the migration file I edited yesterday haha. I will add that to the commit so that it will build the table properly.

As of now I couldn't find any test unit that checks all the props one by one for `GET /api/audios/:id` (there is one for `GET /api/audios/` though in `audiosController.test.js`). Maybe there isn't one at all? 🤔 If you know of any, let me know and I'll commit again!

File: ui/src/components/dashboard/audio_list_item.vue (hunk 2 above)
Comment (austindebruyn) [reply ui/src/components/dashboard/audio_list_item.vue]: it doesn't matter a whole lot, but there's a case to be made for only using css classes for what they're meant for (styling components) when they're needed and using plain divs for what they're meant for (organizing content).

**Pull request merged**